cloudinit/sources/DataSourceOracle.py

```diff
@@ -110,7 +124,7 @@ class DataSourceOracle(sources.DataSource):
         sources.NetworkConfigSource.SYSTEM_CFG,
     )
 
-    _network_config = sources.UNSET
+    _network_config: Optional[dict] = None
```

Comment:
What if we did something like
```python
_network_config: dict = {"config": []}
```

and instead of checking for None, we have a function to check if the config list has items. That way, we avoid casts and/or None type errors.

Reply:
Done.